### PR TITLE
Compliance and Compliance Score Filter

### PR DESCRIPTION
**Pull request by dLobatog**
Peek 2019-04-10 14-17

Requires https://github.com/RedHatInsights/compliance-backend/pull/104#pullrequestreview-225109332 to work

### CODE REVIEW

File: src/SmartComponents/SystemsComplianceFilter/SystemsComplianceFilter.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+`;
+
+class SystemsComplianceFilter extends React.Component {
+    state = {
+        compliant: undefined,
+        complianceScore: []
+    };
+
+    buildFilter = () => {
```
Comment (akofink) [inline src/SmartComponents/SystemsComplianceFilter/SystemsComplianceFilter.js]: `const`? (and the other methods)

File: src/SmartComponents/SystemsComplianceFilter/SystemsComplianceFilter.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import gql from 'graphql-tag';
+
+const GET_SYSTEMS = gql`
+query getSystems($filter: String!) {
+    allSystems(search: $filter) { id }
+}
+`;
+
+class SystemsComplianceFilter extends React.Component {
+    state = {
```
Comment (akofink) [inline src/SmartComponents/SystemsComplianceFilter/SystemsComplianceFilter.js]: should we use default props instead?

File: src/SmartComponents/SystemsTable/SystemsTable.js [hunk 3]
```diff
@@ -14,14 +15,23 @@ class SystemsTable extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
```
Comment (akofink) [inline src/SmartComponents/SystemsTable/SystemsTable.js]: default props here as well (I know, it'd be a partial refactoring)?

File: package.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
     "@patternfly/react-core": "2.10.6",
     "@patternfly/react-icons": "3.7.1",
     "@patternfly/react-table": "1.4.6",
-    "@red-hat-insights/insights-frontend-components": "3.41.5",
+    "@red-hat-insights/insights-frontend-components": "3.41.12",
```
[the remark this answers is not in the document]
Comment (dLobatog) [reply package.json]: A while ago changes in the insights-frontend-components repo broke things like tabs, tables, etc.. without warning, updating one version at a time has 0 cost and ensures we know what version we need to not break anything

File: src/SmartComponents/SystemsComplianceFilter/SystemsComplianceFilter.js (hunk 1 above)
Comment (dLobatog) [reply src/SmartComponents/SystemsComplianceFilter/SystemsComplianceFilter.js]: I'm not sure what you mean? `const` isn't required to define methods in a React component, in fact `const buildFilter` makes webpack unable to compile this (`unexpected token.`)

File: src/SmartComponents/SystemsTable/SystemsTable.js (hunk 3 above)
Comment (dLobatog) [reply src/SmartComponents/SystemsTable/SystemsTable.js]: Default props for what?

**Comment by dLobatog** [comment]
Peek 2019-04-11 14-28

File: src/SmartComponents/SystemsComplianceFilter/SystemsComplianceFilter.js (hunk 1 above)
Comment (akofink) [reply src/SmartComponents/SystemsComplianceFilter/SystemsComplianceFilter.js]: Ok :man_shrugging: I don't understand `const` lol. I thought it just made the pointer to the function constant, but I'm not sure why you can't use it here. You typically see anonymous functions declared like `const foo = () => {}`. Maybe in a class, it doesn't make sense?

File: src/SmartComponents/SystemsTable/SystemsTable.js (hunk 3 above)
Comment (akofink) [reply src/SmartComponents/SystemsTable/SystemsTable.js]: For this SystemsTable component. We currently don't have any default props defined.

File: src/SmartComponents/SystemsTable/SystemsTable.js [hunk 5]
```diff
@@ -16,6 +17,7 @@ class SystemsTable extends React.Component {
...
             InventoryCmp: () => <EmptyTable><Spinner/></EmptyTable>,
             items: this.props.items,
+            filterEnabled: false
```
Comment (akofink) [inline src/SmartComponents/SystemsTable/SystemsTable.js]: syntax error here

```
ERROR in ./src/SmartComponents/SystemsTable/SystemsTable.js                                                                                                                                                                                                                                           
Module build failed (from ./node_modules/babel-loader/lib/index.js):                                                                                                                                                                                                                                  
SyntaxError: /home/akofink/dev/repos/compliance-frontend/src/SmartComponents/SystemsTable/SystemsTable.js: Unexpected token, expected "," (21:12)                                                                                                                                                     
                                                                                                                                                                                                                                                                                                      
  19 |             items: this.props.items,                                                                                                                                                                                                                                                           
  20 |             filterEnabled: false                                                                                                                                                                                                                                                               
> 21 |             meta: {
     |             ^
  22 |                 page: 1,
  23 |                 perPage: 50,
  24 |                 totalItems: this.props.items.length
```

**Comment by akofink** [comment]
I'm getting this warning. Do you know how we can remove it?

> A component is changing an uncontrolled input of type checkbox to be controlled. Input elements should not switch from uncontrolled to controlled

**Comment by akofink** [comment]
Also, this checkbox doesn't make sense:

image

**Comment by akofink** [comment]
Another comment: Is this a OR or AND checkbox? I can check compliant and non-compliant, and it shows  compliant systems, but if I check non-compliant and 90-100%, I see no systems. It seems like if you check all the boxes, it's `(compliant | non-compliant) & (90-100 | 70-89 | 50-69 | < 50)`

**Pull request merged**